zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -735,17 +798,22 @@ impl<F: Field> MPTConfig<F> {
                 memory.clear_witness_data();
 
                 let mut offset = 0;
-                for node in nodes.iter() {
+                let mut offset_prev = 0;
+                for (node_id, node) in nodes.iter().enumerate() {
                     // Assign bytes
                     for (idx, bytes) in node.values.iter().enumerate() {
                         for (byte, &column) in bytes.iter().zip(self.main.bytes.iter()) {
                             assign!(region, (column, offset + idx) => byte.scalar())?;
                         }
+                        assign!(region, (self.q_row, offset + idx) => idx.scalar())?;
+                        assign!(region, (self.q_node, offset + idx) => offset.scalar())?;
+                        assign!(region, (self.q_node_prev, offset + idx) => offset_prev.scalar())?;
```

Comment:
Keeping the total row idx in `q_node` + `q_row` is possible, but I don't see that much benefit over it compared to always starting from `0` or `RowType::Count` for each row? I guess it's quite similar, but seems less state to keep track of that way.

Reply:
Yeah, all my efforts was for constraining `Start -> Branch* -> Account -> Branch* -> Storage` with their designated rows offsets. But since you explained that the circuits have these constraint already with `is_below_account`, I guess these are not necessary...